AK/Bitmap.h

```diff
@@ -157,6 +157,47 @@ class Bitmap {
         return -1;
     }
 
+    int find_up_to_n_unset(int n, int& region_size) const
+    {
+        /* return this in case no more that one unset bits were found */
+        int first_index = find_first_unset();
+        if (first_index == -1)
+            return -1;
+
+        int free_region_start = first_index;
+        int free_region_size = 1;
+
+        int max_region_start = free_region_start;
+        int max_region_size = free_region_size;
+
+        /* Let's try and find the best fit possible */
+        for (int j = first_index + 1; j < m_size && free_region_size < n; j++) {
+            if (!get(j)) {
+                if (free_region_size == 0)
+                    free_region_start = j;
+                free_region_size++;
+            } else {
+                if (max_region_size < free_region_size) {
+                    max_region_size = free_region_size;
+                    max_region_start = free_region_start;
+                }
+                free_region_start = 0;
+                free_region_size = 0;
+            }
+        }
+
+        if (max_region_size < free_region_size) {
+            max_region_size = free_region_size;
+            max_region_start = free_region_start;
+        }
+        /* Set the output region size */
+        region_size = max_region_size;
+        if (max_region_size > 1)
+            return max_region_start;
+        /* if the max free region size is one, then return the earliest one ??*/
```

Comment:
Please use C++ style comments (`//`).
Also, is there a question here that we need to answer? What do you mean by `??` in this context?

Reply:
I was not sure if I want to return the earliest block or the latest one found. I ended up returning the earliest but this increases the look up time for the next first unset block.